xbmc/settings/dialogs/GUIDialogLibExportSettings.cpp

```diff
@@ -0,0 +1,376 @@
+/*
+*      Copyright (C) 2005-2014 Team XBMC
+*      http://xbmc.org
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with XBMC; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <limits.h>
+
+#include "GUIDialogLibExportSettings.h"
+#include "dialogs/GUIDialogFileBrowser.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "messaging/helpers/DialogHelper.h"
+#include "messaging/helpers/DialogOKHelper.h"
+#include "ServiceBroker.h"
+#include "settings/SettingUtils.h"
+#include "settings/lib/Setting.h"
+#include "settings/Settings.h"
+#include "settings/windows/GUIControlSettings.h"
+#include "storage/MediaManager.h"
+#include "Util.h"
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "filesystem/Directory.h"
+
+#define SETTING_MUSIC_EXPORT_FILETYPE "musiclibrary.exportfiletype"
+#define SETTING_MUSIC_EXPORT_FOLDER "musiclibrary.exportfolder"
+#define SETTING_MUSIC_EXPORT_ITEMS "musiclibrary.exportitems"
+#define SETTING_MUSIC_EXPORT_UNSCRAPED "musiclibrary.exportunscraped"
+#define SETTING_MUSIC_EXPORT_OVERWRITE "musiclibrary.exportoverwrite"
+#define SETTING_MUSIC_EXPORT_ARTWORK "musiclibrary.exportartwork"
+#define SETTING_MUSIC_EXPORT_SKIPNFO "musiclibrary.exportskipnfo"
```

Comment:
Please use `const` instead of `#define`

Reply:
@xhaggi I used `#define` to be consistent with other dialogs e.g. `CGUIDialogAudioSubtitleSettings`, `CGUIDialogNetworkSetup`, `GUIDialogPVRTimerSetting`, `CGUIDialogVideoSettings` etc. I could not find anywhere that used `const` in a similar context. Maybe all need to change, or is `#define` OK?

Reply:
We should prefer const over define for newer code. Same with cast and c++ style cast.

Reply:
These other dialogs were all visited recently (#12251), but `#define` not replaced? Sorry to be quibbling, but this just seems an inconsistent change to make.

Reply:
I assume what you are suggesting is replace them with

Reply:
After a second look you should move them to `CSettings` h/cpp

Reply:
That makes some sense. I have to say that it has been far from obvious how a dialog that remembers previous values should be best implemented. I have tried to follow existing examples, with varied success.

Reply:
Adding the export settings to to media settings -> music -> export should not be an issue if you asked me. We do that for various other settings too.

Reply:
Media settings are a bit of a jumble - some handled by CServiceBroker, others written by CMediaSettings directly to XML. The previous export criteria only want displaying when you are about to export, so don't belong on the media settings screen itself. Coming up on a dialog is correct. But I will follow the PVR settings (most recent addition I think) and switch to `const` in CSettings as you suggest.